prime-router/docs/openapi.yml

```diff
@@ -9,6 +9,399 @@ info:
   version: 0.2.0-oas3
 paths:
 
+  # The sender end-point handles receive actions.
+  #
+  # END-POINTS
+  # /submissions
+  # /submissions/multorg
+  # /submissions/{reportid}
+  # /submissions/{organizationName}
+  #
+  # FILTERS (OPTIONAL)
+  # ?page=INT
+  # ?pagesize=INT
+  # ?getLength=BOOLEAN
+  # ?dateStart=STRING (YYYY-MM-DD) || INT (NN [days back] | default: 30)
+  # ?dateEnd=STRING (YYYY-MM-DD | default: Today)
+  # ?year=INT (YYYY | default: ALL)
+  # ?month=INT (MM | default: ALL | requires ?year)
+  # ?organizations=INT (default: ALL)
+  # ?recipients=ARRAY
+  # ?sender=ARRAY
+  # ?warnings=STRING
+  # ?errors=STRING
+  # ?receipts=STRING
+  #
+  #
+  #
+  #
+  # 30 days (based on where request orignates)
+  # pagination?
+  # admin filter on org
+  # filter selections (date range, column headers, etc.)
+  # records of what users have done?
+  # PDF exports?
+  # Org admin, multi-org aggregate
+  #
+  # 
+  # /REJECTION:
+  # Submission Date | Filename | Reason(s) of Rejection
+  # 
+  # /Confirmation of Acceptance
+  # 
+  # 
+  # 
+  #
+  /submissions:
+    get:
+      description: Return a list of all basic info on all receive actions and associated reports. Must have admin access.
+      security:
+        - OAuth2: [ system_admin ]
+        - ApiKeyAuth: []
+      parameters:
+        - in: query
```

Comment:
In general, offset pagination doesn't work well for APIs for logs and other data that are changing. I suggest that you change this to a cursor-based pagination scheme. There are many articles explaining the reasoning, but this StackOverflow post explains this a little bit. https://stackoverflow.com/questions/18314687/how-to-implement-cursors-for-pagination-in-an-api. For examples of cursor-based APIs consider the Slack or Okta APIs.

Reply:
Agreed that cursors are a more robust approach for data that changes frequently and can be very large (there's a reason DBs use them).

My primary focus is on end-user usability that these features expose since I'm on the client side. Cursor based client UI lends itself to things like cloud log UIs and "infinite scrolling" lists. Often, to find anything useful with this approach, search and/or filter features are required from the start (because the user cannot jump to "page N"). It's typically less novice-user-friendly.

(Speaking of Search vs Filter: are there any expectation that deep searches might become a requirement versus just filtering on the metadata?)

We can drop my suggestion of using pagination with offsets and page sizes and having a call to get the total length.

We should plan the React client's features accordingly. Filtering by date will probably be required much sooner. I can discuss our UX options with @drew-usds